UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java

```diff
@@ -145,6 +117,57 @@ public void onFailure(IMqttToken asyncActionToken, Throwable exception) {
         });
     }
 
+    /**Driving using thumbstick
+     * If cruise control is enabled, speed will be set from seekBar's progress
+     * If cruise control is disabled, speed will be set as percentage of seekBar's progress*/
+    @Override
+    public void onThumbstickMoved(float xPercent, float yPercent, int id) {
+        int angle = (int)((xPercent) * 100);
+        int strength;
+        //TODO: check why do we need the negative of seekBar.getProgress()
+        int seekProgress = - seekBar.getProgress();
```

Comment:
ok, I see you removed the listener and are just retrieving the speed when activating the cruise control or not. 
And retrieving the value when moving the joystick ! Good job, we had a lot of thinking around that haha